src/components/heading/heading.twig

```diff
@@ -20,15 +20,21 @@
   Build the class name based on the level.
 #}
 
-{% set _level_class = 'c-heading' %}
+{% set _heading_class = 'c-heading' %}
+{% set _heading_class_prefix = 'c-heading' %}
 
 {% if level is not empty %}
   {% set _level_modifier = ('level-' ~ level)|replace({ '--': '-n' }) %}
-  {% set _level_class = _level_class ~ ' ' ~ _level_class ~ '--' ~ _level_modifier %}
+  {% set _heading_class = _heading_class ~ ' ' ~ _heading_class_prefix ~ '--' ~ _level_modifier %}
+{% endif %}
+
+{% if weight is not empty %}
+  {% set _weight_modifier = ('weight-' ~ weight) %}
```

Comment:
It looks like you updated the class to just be `--light`, but you're still prepending `weight-` in the template. This whole block could probably be simplified now. Maybe you just need a `true` or `false` value for `light`? Or maybe you don't need a new option at all and can just pass along the `class` of `c-heading--light`?